includes/modules/payment/securesubmit.php

```diff
@@ -164,50 +174,42 @@ public function process_button() {
 
     public function before_process() {
         global $_POST, $order, $sendto, $currency, $charge, $db, $messageStack; 
-        require_once(DIR_FS_CATALOG . DIR_WS_MODULES . 'payment/securesubmit/Hps.php');
+        require 'vendor/autoload.php';
         $error = '';
         
-        $config = new HpsConfiguration();
-        $config->secretApiKey = MODULE_PAYMENT_SECURESUBMIT_SECRET_KEY;
-        $config->versionNumber = '1512';
-        $config->developerId = '002914';
+        $chargeService = $this->getConfig();
 
-        $chargeService = new HpsCreditService($config);
-
-        $hpsaddress = new HpsAddress();
+        $hpsaddress = new Address();
         $hpsaddress->address = $order->billing['street_address'];
         $hpsaddress->city = $order->billing['city'];
         $hpsaddress->state = $order->billing['state'];
         $hpsaddress->zip = preg_replace('/[^0-9]/', '', $order->billing['postcode']);
         $hpsaddress->country = $order->billing['country']['title'];
 
-        $cardHolder = new HpsCardHolder();
+        $cardHolder = new Customer();
         $cardHolder->firstName = $order->billing['firstname'];
         $cardHolder->lastName = $order->billing['lastname'];
         $cardHolder->emailAddress = $order->customer['email_address'];
         $cardHolder->address = $hpsaddress;
 
-        $hpstoken = new HpsTokenData();
-        $hpstoken->tokenValue = $_POST['securesubmit_token']; 
+        $hpstoken = new CreditCardData();
+        $hpstoken->token = $_POST['securesubmit_token']; 
 
         $last_order = $db->Execute("select orders_id from " . TABLE_ORDERS . " order by orders_id desc limit 1");
         $this->invoice_number = ($last_order->fields['orders_id']) + 1;
-        
 
-        $details = new HpsTransactionDetails();
+        $details = new TransactionSummary();
         $details->invoiceNumber = $this->invoice_number;
-
         try {
             if (MODULE_PAYMENT_SECURESUBMIT_AUTHCAPTURE == 'Authorize') {
-                $response = $chargeService->authorize(
+                $response = $hpstoken->authorize(
                         round($order->info['total'], 2), MODULE_PAYMENT_SECURESUBMIT_CURRENCY, $hpstoken, $cardHolder, false, null
                 );
```

Comment:
The `authorize` method on this object doesn't support this signature. Let's update it to use the fluent methods on the builder.

Reply:
updated